stablehlo/dialect/TypeInference.cpp

```diff
@@ -369,6 +400,40 @@ LogicalResult verifyAddOp(std::optional<Location> location, Operation* op,
   return success();
 }
 
+LogicalResult verifyTransposeOp(std::optional<Location> location,
+                                Type operandType, ArrayRef<int64_t> permutation,
+                                Type resultType) {
+  // transpose_c1
+  if (allQuantized<quant::UniformQuantizedType>(operandType, resultType)) {
+    if (getElementTypeOrSelf(operandType) != getElementTypeOrSelf(resultType))
+      return emitOptionalError(location,
+                               "expect same quantization scale and zero_point "
+                               "for operand and result but got ",
+                               operandType, " vs ", resultType);
+  }
+  if (allQuantized<quant::UniformQuantizedPerAxisType>(operandType,
+                                                       resultType)) {
```

Comment:
Is it valid for transpose to have a quantized operand but not a quantized result? Is this handled elsewhere?

Reply:
HLO_CompatibleOperandsAndResultElementType Trait is handling quant vs non quant or per-tensor vs per-axis mismatch.